Sources/SwiftProtobuf/ProtobufJSONEncoding.swift

```diff
@@ -21,16 +21,32 @@ struct ProtobufJSONEncodingVisitor: ProtobufVisitor {
     private var encoder = ProtobufJSONEncoder()
     var result: String {return encoder.result}
 
-    init() {}
+    private var nameResolver: (Int) -> String?
+    private var anyTypeURL: String?
+
+    init(message: ProtobufJSONMessageBase, anyTypeURL: String? = nil) throws {
+        self.nameResolver =
+            ProtoNameResolvers.jsonFieldNameResolver(for: message)
+        self.anyTypeURL = anyTypeURL
 
-    init(message: ProtobufJSONMessageBase) throws {
         try withAbstractVisitor {(visitor: inout ProtobufVisitor) in
             try message.traverse(visitor: &visitor)
         }
     }
 
     mutating func withAbstractVisitor(clause: (inout ProtobufVisitor) throws -> ()) throws {
         encoder.startObject()
+
+        // TODO: This is a bit of a hack that exists as a workaround to make the
+        // hand-written Any serialization work with the new design. We need to
+        // generate those WKTs instead of maintaining the hand-written ones,
+        // handle the special cases differently, and then remove this.
```

Comment:
I've started thinking of 'Any JSON' serialization as its own distinct serialization different from 'regular JSON'.  The two obviously have a lot in common, but they're not the same.